File: app/gui/design/tokens/elevation.py
"""
SOC-IQ Design System
Elevation Design Tokens

Defines semantic elevation levels for the application.

Elevation describes visual hierarchy rather than specific shadow
implementations. The actual shadow effects will be created later
by the design effects layer.
"""

from __future__ import annotations


class Elevation:
    """Semantic elevation levels."""

    # Flat surfaces
    NONE = 0

    # Cards placed on the page
    LOW = 1

    # Elevated panels and hover states
    MEDIUM = 2

    # Floating widgets
    HIGH = 3

    # Dialogs
    DIALOG = 4

    # Notifications / popups / tooltips
    OVERLAY = 5


class ZIndex:
    """Logical stacking order."""

    BACKGROUND = 0
    CONTENT = 100
    SIDEBAR = 200
    HEADER = 300
    DROPDOWN = 500
    TOOLTIP = 700
    DIALOG = 900
    NOTIFICATION = 1000

File: frontend/src/styles/tokens/elevation.ts
/**
 * SOC-IQ Design System — Elevation Tokens
 * Ported 1:1 from `app/gui/design/tokens/elevation.py`.
 *
 * Elevation describes visual hierarchy (a level), not a specific
 * box-shadow implementation — the shadow values themselves live in
 * `shadow` (see `color.ts`) and are composed at the component layer,
 * same as the Python source's own stated intent.
 */

export const Elevation = {
  none: 0,
  low: 1,
  medium: 2,
  high: 3,
  dialog: 4,
  overlay: 5,
} as const;

export const ZIndex = {
  background: 0,
  content: 100,
  sidebar: 200,
  header: 300,
  dropdown: 500,
  tooltip: 700,
  dialog: 900,
  notification: 1000,
} as const;

export type ElevationToken = typeof Elevation;
export type ZIndexToken = typeof ZIndex;
